compiler/src/iree/compiler/Codegen/LLVMCPU/LLVMCPULowerExecutableTarget.cpp

```diff
@@ -158,7 +158,7 @@ static TilingConfig getTilingConfigForPipeline(ModuleOp moduleOp) {
 void LLVMCPULowerExecutableTargetPass::runOnOperation() {
   IREE::HAL::ExecutableVariantOp variantOp = getOperation();
   ModuleOp moduleOp = variantOp.getInnerModule();
-  if (!variantOp || !moduleOp) {
+  if (!moduleOp) {
```

Comment:
@benvanik should this be made part of the variantOp verification instead?

Reply:
modules are optional as external variants (linking in external object files) have no modules - omitting the module should effectively disable codegen

Reply:
Based on https://github.com/openxla/iree/pull/15277#discussion_r1370773874 maybe we just return silently here... For now an error is ok though.

Reply:
good point - well, this PR doesn't change the behavior, but we should look out for that as people start pushing in external dispatches (mostly GPU to start, as ukernels are best for CPU and those will have modules) - there must be something higher up opting out of these passes but it'd be good to have defense in depth